sn_networking/src/bootstrap.rs

```diff
@@ -124,23 +138,39 @@ impl ContinuousBootstrap {
         // kad bootstrap process needs at least one peer in the RT be carried out.
         let should_bootstrap = !self.is_ongoing && peers_in_rt >= 1;
 
+        // if we have less than DESIRABLE_BUCKET_COUNT buckets, then we should bootstrap
+        if bucket_stats.len() < DESIRABLE_BUCKET_COUNT {
+            info!("We have less than {DESIRABLE_BUCKET_COUNT} buckets. Continuing to bootstrap.");
+            return (should_bootstrap, None);
+        }
+
+        // If we're at a DESIRABLE_BUCKET_COUNT, we can think about slowing down the bootstrapping process
         // if it has been a while (LAST_PEER_ADDED_TIME_LIMIT) since we have added a new peer to our RT, then, slowdown
         // the bootstrapping process.
         // Don't slow down if we haven't even added one peer to our RT.
-        if self.last_peer_added_instant.elapsed() > LAST_PEER_ADDED_TIME_LIMIT && peers_in_rt != 0 {
+        if self.last_peer_added_instant.elapsed() > LAST_PEER_ADDED_TIME_LIMIT {
             info!(
-                "It has been {LAST_PEER_ADDED_TIME_LIMIT:?} since we last added a peer to RT. Slowing down the continuous bootstrapping process"
+                "It has been {LAST_PEER_ADDED_TIME_LIMIT:?} since we last added a peer to RT. Increasing the rate of continuous bootstrapping"
             );
 
-            let mut new_interval = tokio::time::interval(NO_PEER_ADDED_SLOWDOWN_INTERVAL);
+            let mut new_interval = tokio::time::interval(BOOTSTRAP_INTERVAL * 2);
             new_interval.tick().await; // the first tick completes immediately
             return (should_bootstrap, Some(new_interval));
         }
 
+        // We want to aim for a decent average bucket size.
+        // we want the average bucket size to be at least K_VALUE / 3
+        if (peers_in_rt / bucket_stats.len()) < DESIRED_AVG_BUCKET_SIZE {
+            info!(
+                "The average bucket is < {DESIRED_AVG_BUCKET_SIZE} peers. Continuing to bootstrap."
+            );
+            return (should_bootstrap, None);
```

Comment:
The code below where we `increment the interval based on the number of peers`, can be triggered prior to this check and could set the interval to a way higher number than desirable.

We should set a new interval inside this check. Something like this
```rust
let interval = if current_interval != AVG_BUCKET_SIZE_INTERVAL {
	let mut new_interval = tokio::time::interval(AVG_BUCKET_SIZE_INTERVAL);
	new_interval.tick().await; // the first tick completes immediately
	Some(new_interval)
} else {
	None
};
return (should_bootstrap, interval)
```

Reply:
wrong commit, updated now